### PR TITLE
Initial docs

### LINKED ISSUE #30: Add initial docs
Add initial documentation to the project.

### PR DESCRIPTION
**Pull request by forman**
Closes #30

### CODE REVIEW

File: chartlets.py/README.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # chartlets 
 
-`chartlets` is a Python framework for server-configured web-UI contributions. 
+Chartlets is a software frameworks that allows websites developed with
```
Comment (b-yogesh) [inline chartlets.py/README.md:3]:
```suggestion
Chartlets is a software framework that allows websites developed with
```

File: docs/components.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Given here is a list of the components supported by chartlets.
+
+Using the Python package, they are be imported from the `chartlets.components`
```
Comment (b-yogesh) [inline docs/components.md:3]:
```suggestion
Using the Python package, they can be imported from the `chartlets.components` 
```

File: chartlets.js/README.md [hunk 3]
```diff
@@ -1,54 +1,16 @@
...
-and plugging them into existing web frontends.
+Chartlets is a software frameworks that allows websites developed with
+React to be extended by server-side widgets programmed in Python or other
+programming languages. 
 
-_Note, this library is experimental and under development still._
+This is the JavaScript/React library of the framework.
 
-## How it is supposed to work
+For details refer to the [documentation](https://bcdev.github.io/chartlets/).
```
Comment (b-yogesh) [inline chartlets.js/README.md:9]: This link doesn't exist yet

File: docs/demo.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+Chartlets provides a simple demo that explains serves as a
```
Comment (b-yogesh) [inline docs/demo.md:1]:
```suggestion
Chartlets provides a simple demo that serves as a
```

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Welcome to Chartlets
+
+Chartlets is a software frameworks that allows websites developed with
```
Comment (b-yogesh) [inline docs/index.md:3]:
```suggestion
Chartlets is a software framework that allows websites developed with
```

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+Users write the widgets in, e.g. Python, and a REST server implements three 
+endpoints to publish the widgets:
+
+- `GET /contributions`: Called once after application UI starts up.
+  Returns an object whose keys are contribution points (e.g., "panels") 
+  and whose values are arrays of contribution objects.
+- `POST /layout/{contribPoint}/{contribIndex}`:
+  Called once for every contribution when it becomes visible in the UI.
+  Returns the contribution's initial component tree
```
Comment (b-yogesh) [inline docs/index.md:17]:
```suggestion
  Returns the contribution's initial component tree.
```

File: docs/usage.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+```python
+from chartlets import Extension
+from chartlets.demo.contribs import Panel
+
+Extension.add_contrib_point("panels", Panel)
+```
+
+### 3. Load the extensions
+
+Load the extensions that augment your application.
```
Comment (b-yogesh) [inline docs/usage.md:41]:
```suggestion
Load the extensions that augments your application.
```

File: docs/usage.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+The chartlets backend implementation is provided by the Python package `chartlets`.
+It makes it easy to implement the chartlet endpoints in your preferred
+webserver framework, such as Flask, FastAPI, or Tornado. 
+
+### 1. Implement the possible contributions
+
+Implement the application-specific contributions that users 
+can add to their extensions.
+
+As an example, see [`panel.py` of the demo](chartlets.py/chartlets/demo/contribs/panel.py):
```
Comment (b-yogesh) [inline docs/usage.md:13]: This applies to all the links below as well, they all return 404. Are they yet to be added in issue https://github.com/bcdev/chartlets/issues/34?

**Review by b-yogesh — CHANGES_REQUESTED** [review]
Looks good. Please have a look at the small change requests.

File: docs/usage.md (hunk 8 above)
Comment (forman) [reply docs/usage.md:13]: I can onlöy link to them once we have deployed the docs on GH pages.

File: docs/usage.md (hunk 7 above)
Comment (forman) [reply docs/usage.md]: Nope. _augment_ is correct, it is a plural.

**Review by b-yogesh — APPROVED** [review]

**Pull request merged**